corehq/form_processor/management/commands/populate_missing_delete_on_values.py

```diff
@@ -0,0 +1,16 @@
+from django.core.management.base import BaseCommand
+
+from corehq.form_processor.models import XFormInstance
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+
+
+class Command(BaseCommand):
+    help = "One time use. Populate missing XFormInstance.deleted_on values"
+
+    def handle(self, **options):
+        for db in get_db_aliases_for_partitioned_query():
+            queryset = XFormInstance.objects.using(db).filter(deleted_on__isnull=True)
+            for xform in queryset:
+                if xform.doc_type == 'XFormInstance-Deleted':
```

Comment:
`xform.doc_type` is a computed property based on the presence of the `deleted_on` field so this won't work as a way to determine if the form 'should' be deleted.

However, since the bug also updated the `state` property, you should be able to use that to determine if the form is deleted or not.

Reply:
Thanks for pointing that out! I switched to using the old deleted state property along with Daniel's suggestion below